Branch-TestBed/Branch-SDK-Tests/BNCClassSerializationTests.m

```diff
@@ -0,0 +1,117 @@
+//
+//  BNCClassSerializationTests.m
+//  Branch-SDK-Tests
+//
+//  Created by Ernest Cho on 3/28/24.
+//  Copyright © 2024 Branch, Inc. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "BranchEvent.h"
+#import "BranchOpenRequest.h"
+#import "BranchInstallRequest.h"
+
+@interface BranchEvent()
```

Comment:
Do we need to test the write to disk portion?

Reply:
I don't think so. This test suite is focused on checking secure coding behavior on a few classes. Everything else is considered out of scope.

Also testing the disk writes would require refactoring and possibly a separate test suite since our tests are run in parallel so a global data source such as disk would cause issues.